pkg/action/scan_error.go

```diff
@@ -0,0 +1,67 @@
+package action
+
+import "fmt"
+
+const (
+	errMsgScanFailed     = "scan failed"
+	errMsgGenerateFailed = "failed to generate file report"
+	errMsgNilReport      = "file report is nil"
+)
+
+type ErrorType int
+
+const (
+	TypeScanError ErrorType = iota
+	TypeGenerateError
+	TypeNilError
```

Comment:
Godoc comments please 🙏

Reply:
Added in [`8e1e5c6` (#777)](https://github.com/chainguard-dev/malcontent/pull/777/commits/8e1e5c6cf378a547f46bccdd7d73ceb9a912eb5b). Let me know if you think any other comments are needed!